examples/third_party/mesa/BUILD.mesa.bazel

```diff
@@ -38,7 +38,21 @@ meson_with_requirements(
             "PATH": "$$(dirname $$EXT_BUILD_ROOT$$/$(location @bison//:gen_dir))/bison/bin:$$(dirname $$EXT_BUILD_ROOT$$/$(location @flex//:flex_exe)):$$EXT_BUILD_DEPS$$/bin/m4/bin/:$$(dirname $$EXT_BUILD_ROOT$$/$(PYTHON3)):$$PATH",
         },
     }),
+    copts = select({
+        # "@openssl//:msvc_compiler": ["/std:c++14"],
+        "@platforms//os:linux": ["-std=c++14"],
+        "//conditions:default": [],
+    }),
+
     lib_source = ":all_srcs",
+    linkopts = select({
+        "@openssl//:msvc_compiler": [
+            "ws2_32.lib",
+        ],
+        "@platforms//os:linux": ["-lstdc++"],
+        # "@platforms//os:macos": ["-mlinker-version=305"],
+        "//conditions:default": [],
+    }),
```

Comment:
I suspect the reason you're having to add these is because CXX is actually passing the c compiler because bazel by default passes `-x c++` explicitly.

Reply:
yes, 
if we don't set CXX/CC, we get `undefined symbol: fcntl64` with zig cc. However, setting them to bazel configured compiler, `/usr/bin/gcc` for both in CI here,  we will need to pass in c++ library